packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Serialization/ScmSerializationInterfaces.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    internal class ScmSerializationInterfaces : SerializationInterfaces
+    {
+        private readonly CSharpType _modelType;
+        private readonly bool _isStruct;
+
+        public ScmSerializationInterfaces(TypeProvider typeProvider, bool hasJson, bool hasXml) : base(typeProvider, hasJson, hasXml)
+        {
+            _modelType = typeProvider.Type;
+            _isStruct = typeProvider.IsStruct;
+
+            if (hasJson)
+            {
+                IJsonModelTInterface = new CSharpType(typeof(IJsonModel<>), _modelType);
+                IPersistableModelTInterface = new CSharpType(typeof(IPersistableModel<>), _modelType);
+                IJsonModelObjectInterface = _isStruct ? (CSharpType)typeof(IJsonModel<object>) : null;
+                IPersistableModelObjectInterface = _isStruct ? (CSharpType)typeof(IPersistableModel<object>) : null;
+            }
+
+            // TO-DO: Handle XML serialization https://github.com/microsoft/typespec/issues/3331
```

Comment:
nit: can we also have a follow up for adding in Bicep serialization?

Reply:
Opened a follow up issues for this based on our discussion: https://github.com/microsoft/typespec/issues/3356